src/org/zaproxy/zap/extension/websocket/alerts/WebSocketAlertWrapper.java

```diff
@@ -0,0 +1,97 @@
+/*
+ * Zed Attack Proxy (ZAP) and its related class files.
+ *
+ * ZAP is an HTTP/HTTPS proxy for assessing web application security.
+ *
+ * Copyright 2018 The ZAP Development Team
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.zaproxy.zap.extension.websocket.alerts;
+
+import org.apache.log4j.Logger;
+import org.parosproxy.paros.Constant;
+import org.parosproxy.paros.core.scanner.Alert;
+import org.parosproxy.paros.model.HistoryReference;
+import org.parosproxy.paros.network.HttpMessage;
+import org.zaproxy.zap.extension.websocket.WebSocketMessageDTO;
+import org.zaproxy.zap.extension.websocket.utility.InvalidUtf8Exception;
+
+
+
+/**
+ * Wrapper for Alerts. This wrapper used to construct alerts for WebSocket
+ */
+public class WebSocketAlertWrapper {
+    
+    private static final Logger LOGGER = Logger.getLogger(WebSocketAlertWrapper.class);
+    
+    private static final String OTHER_INFO_LABEL = "[WebSocket Message] ";
+    
+    private Alert alert;
+    
+    /**
+     * Initialize a new {@link Alert}.
+     * @see {@link Alert#Alert(int, int, int, String)}
+     */
+    public WebSocketAlertWrapper(int pluginId, int risk, int confidence, String name) {
+        alert = new Alert(pluginId, risk, confidence, name);
+    }
+    
+    /**
+     * Sets the details of the Alert
+     * @see {@link Alert#setDetail(String, String, String, String, String, String, String, HistoryReference)}
+     *
+     * @param description the description of the alert
+     * @param param the attack that triggers the issue
+     * @param attack the attack that triggers the issue
```

Comment:
As discussed elsewhere the attack field needs to be dropped or handled. Looks like templates were updated but not the actually java bits.

Reply:
The WebSocketAlertWrapper is a generic class which is used by all types of WebSocket Scans. This means may the  WebSocketAlertWrapper class is used to raise alert triggered by an active scanner. That was the concept behind this. What do you think, should I change this method to used only by passive scanner?

Reply:
Nope I guess I'm good with that. I haven't had a chance to pull the recent modifications and test it, as long as the scripts don't fail to match the function definition (a dummy value is passed or default used..) then I think that's fine.

I guess I kind of jumped to a conclusion on this. Feel free to mark it resolved if I'm wrong.

Reply:
`ScriptsWebSocketPassiveScanner.AlertRaiser.raiseAlert` pass `null`  to `WebSocketAlertWrapper`'s attack parameter by default.